src/base/utils/fs.cpp

```diff
@@ -128,17 +132,26 @@ bool Utils::Fs::smartRemoveEmptyFolderTree(const QString &path)
               , [](const QString &l, const QString &r) { return l.count('/') > r.count('/'); });
 
     for (const QString &p : asConst(dirList)) {
-        // remove unwanted files
-        for (const QString &f : deleteFilesList) {
-            forceRemove(p + f);
-        }
-
-        // remove temp files on linux (file ends with '~'), e.g. `filename~`
         const QDir dir(p);
+        if (!dir.isEmpty(QDir::AllDirs | QDir::NoDotAndDotDot))
+            continue;
         const QStringList tmpFileList = dir.entryList(QDir::Files);
+
+        // deleteFilesList contains unwanted files, usually created by the OS
+        // temp files on linux usually end with '~', e.g. `filename~`
+        const bool hasOtherFiles = !std::all_of(tmpFileList.cbegin(), tmpFileList.cend(), [&deleteFilesList](const QString &f)
```

Comment:
Pardon me, I still think it can be improved a little bit:
```c++
// use `std::any_of()` instead of `!std::all_of()`
const bool hasOtherFiles = std::any_of(tmpFileList.cbegin(), tmpFileList.cend(), [&deleteFilesList](const QString &f)
{
  // cheaper check goes first
  return (!f.endsWith('~') && !deleteFilesList.contains(f, Qt::CaseInsensitive));
});
```

Reply:
Ok, I'll do the changes a bit later. But I would suppose that all_of() would have this optimization inside it. (aka break early on the 1st false).